### ISSUE TITLE
New version (2023.08.07)

### ISSUE THREAD

**Issue by benpye** [Seneral/riscv-openocd-wch#1]
Hey

I got a new version from MRS today - I've stuck it up in a branch myself but you might want to mirror it here as you have some history already. https://github.com/benpye/openocd/commits/mrs-2023-08-07

It seems to be based on a newer riscv-openocd upstream and seems to fix some of the brokeness earlier versions had.

On macOS I need to configure with `./configure --disable-werror --enable-wlinke --disable-ch347` though I think on Windows and Linux `--enable-ch347` might work.

**Comment by Seneral** [comment]
Nice, thanks. Do you mind doing a PR or should I add it myself like the others? (for credit only, I don't mind adding)

**Comment by benpye** [comment]
I don't mind either way - given the bump in the upstream maybe it makes sense to push directly to a new branch?

**Comment by Seneral** [comment]
Right, missed that. Might try a quick and dirty rebase of my current repo to that upstream commit, so what we get is also a difference to their previous version. Not today though

**Comment by Seneral** [comment]
Finally got around to do it, and turns out they might have completely rewritten / restructured their code so it's a lot less "invasive". So didn't make sense to rebase the older versions, instead the new versions diff is better compared to the new riscv-openocd base of 28-10-2022.

**Comment by Seneral** [comment]
Turns out this new version replaced JTAG with SDI - and this is only supported by WCH-LinkE hardware.
Old WCH-Links are not supported with this version anymore.
